### PR TITLE
misc: add 6220 Maestro bin range support, drop pattern regex

### PR DESCRIPTION
**Pull request by mduvall**
This adds the range that is requested in https://github.com/stripe/jquery.payment/issues/157, the exhaustive BIN ranges can be found here (http://www.barclaycard.co.uk/business/files/bin_rules.pdf)—aren't quite close to implementing those in current `jquery.payment` so addressed #157 directly.

r? @SlexAxton

### CODE REVIEW

File: src/jquery.payment.coffee [hunk 1]
```diff
@@ -104,7 +111,10 @@ $.payment.cards = cards = [
 
 cardFromNumber = (num) ->
   num = (num + '').replace(/\D/g, '')
-  return card for card in cards when card.pattern.test(num)
+  for card in cards
+    for pattern in card.patterns
+      r = new RegExp("^#{pattern}")
+      return card if r.exec(num)?
```
Comment (jamesreggio) [inline src/jquery.payment.coffee]: This is too heavy a hammer. If you want to test that the string begins with a pattern, you should use `str.indexOf(pattern) === 0` instead.

Comment (SlexAxton) [inline src/jquery.payment.coffee]: `str.substr(pattern.length) === pattern` gets you the most points

Comment (SlexAxton) [inline src/jquery.payment.coffee]: I assume jdalton did his homework here: https://github.com/lodash/lodash/blob/master/lodash.js#L10854-L10858

**Comment by SlexAxton** [comment]
This felt more like a combinatorics and regex quiz, but it feels fine. I agree with @jamesreggio in getting rid of an unnecessary regex for `startsWith`.

File: src/jquery.payment.coffee (hunk 1 above)
Comment (jamesreggio) [inline src/jquery.payment.coffee]: Well, look at you Mr Smartypants.

Comment (SlexAxton) [inline src/jquery.payment.coffee]: Mr Rangio, that kind langauge will not be toleraited.

Comment (jamesreggio) [inline src/jquery.payment.coffee]: UNSUBSCTIBE

**Pull request merged**